### PR TITLE
Pmpsupdate

### PR DESCRIPTION
**Pull request by tongju12**
## New PMPS system in TMO motion without SP1K4 PMPS

Adding new PMPS library and fill in all states of each motor that matches the name is PMPS database
## Motivation and Context
 It is for SC in TMO and will let scientists change the beam class from database

https://jira.slac.stanford.edu/browse/LCLSECSD-1964
## How Has This Been Tested?

Test each target from every motor and check PMPS GUI

## Where Has This Been Documented?

https://confluence.slac.stanford.edu/pages/viewpage.action?pageId=373331167

## Pre-merge checklist
- [ x] Code works interactively
- [ ] Code contains descriptive comments
- [x ] Test suite passes locally
- [ ] Libraries are set to fixed versions and not ``Always Newest``
- [ ] Code committed with ``pre-commit`` (alternatively ``pre-commit run --all-files``)

### CODE REVIEW

File: plc-tmo-motion/tmo_motion/POUs/TM2K4/ENUM_TM2K4_States.TcDUT [hunk 1]
```diff
@@ -11,7 +11,8 @@ TYPE ENUM_TM2K4_States :
     TARGET1 := 2,
     TARGET2 := 3,
     TARGET3 := 4,
-    TARGET4 := 5
+    TARGET4 := 5,
+	TARGET5 := 6
```
Comment (ZLLentz) [inline plc-tmo-motion/tmo_motion/POUs/TM2K4/ENUM_TM2K4_States.TcDUT:15]: Small question: why are these called TARGET4 and TARGET5 on the ATMs in the enum here, but YAG and DIODE in the database states? You could always rename these here if you wanted.

File: plc-tmo-motion/tmo_motion/POUs/PRG_TM2K4.TcPOU [hunk 2]
```diff
@@ -15,44 +15,47 @@ VAR
...
 fbTM2K4.stTarget4.bUseRawCounts := FALSE;
-fbTM2K4.stTarget4.nRequestAssertionID := 16#5245;
-fbTM2K4.stTarget4.stBeamParams := PMPS_GVL.cstFullBeam;
 fbTM2K4.stTarget4.bValid := TRUE;
+fbTM2K4.stTarget4.stPMPS.sPmpsState := 'TM2K4:ATM-YAG';
+
+fbTM2K4.stTarget5.fPosition := -69;
+fbTM2K4.stTarget5.bUseRawCounts := FALSE;
+fbTM2K4.stTarget5.bValid := TRUE;
+fbTM2K4.stTarget5.stPMPS.sPmpsState := 'TM2K4:ATM-DIODE';
```
Comment (ZLLentz) [inline plc-tmo-motion/tmo_motion/POUs/PRG_TM2K4.TcPOU:48]: Hi Tong, did target 5 here (diode) load and run correctly in the PLC? It doesn't show up in my diagnostic in the IOC, so either there was a load issue or something weird happened with the IOC build. Let me know if I should debug the diagnostic GUI.
image

**Review by ZLLentz — CHANGES_REQUESTED** [review]
Hi Tong, I found one typo in IM3K4 and some oddity in TM2K4's loading as seen in the client GUI. The rest looked good. I had some additional notes and small questions.

File: plc-tmo-motion/tmo_motion/POUs/PRG_1_PlcTask.TcPOU [hunk 3]
```diff
@@ -24,7 +23,7 @@ PRG_ST4K4_TMO_TERM();
 //PRG_ST5K4_TMO_TERM_FIXED();
 PRG_TM1K4();
 PRG_TM2K4();
-PRG_SP1K4();
+//PRG_SP1K4();
```
[the remark this answers is not in the document]
Comment (tongju12) [reply plc-tmo-motion/tmo_motion/POUs/PRG_1_PlcTask.TcPOU:26]: Yes, TMO knew it and it is ok with us now.

File: plc-tmo-motion/tmo_motion/POUs/PRG_IM3K4_PPM.TcPOU [hunk 4]
```diff
@@ -23,41 +23,39 @@ VAR
...
 fbIM3K4.stYag1.bValid := TRUE;
+fbIM3K4.stYag1.stPMPS.sPmpsState := 'IM3K4:PPM-YAG1';
 
 fbIM3K4.stYag2.fPosition := -94.93;
 fbIM3K4.stYag2.fVelocity := fStartupVelo;
 fbIM3K4.stYag2.bUseRawCounts := FALSE;
-fbIM3K4.stYag2.nRequestAssertionID := 16#2344;
-fbIM3K4.stYag2.stBeamParams := PMPS_GVL.cstFullBeam;
 fbIM3K4.stYag2.bValid := TRUE;
+fbIM3K4.stYag1.stPMPS.sPmpsState := 'IM3K4:PPM-YAG2';
```
[the remark this answers is not in the document]
Comment (tongju12) [reply plc-tmo-motion/tmo_motion/POUs/PRG_IM3K4_PPM.TcPOU]: It was a mistake. I fix it in the new commit. Thanks!

File: plc-tmo-motion/tmo_motion/POUs/PRG_TM2K4.TcPOU (hunk 2 above)
Comment (tongju12) [reply plc-tmo-motion/tmo_motion/POUs/PRG_TM2K4.TcPOU:48]: Yes, TM2K4 targt 5 is working well in PLC and PMPS main screen can watch it while it is moving but somehow this client GUI does not show target 5 though.

Comment (ZLLentz) [reply plc-tmo-motion/tmo_motion/POUs/PRG_TM2K4.TcPOU:48]: I'm creating an issue in my repo to investigate this later: https://github.com/pcdshub/pmpsdb_client/issues/20

**Review by ZLLentz — APPROVED** [review]
Updated from suggestions and the important CI jobs pass :+1:

**Pull request merged**